src/main/java/spoon/support/reflect/declaration/CtTypeImpl.java

```diff
@@ -739,6 +740,48 @@ private boolean isSameParameter(CtTypeReference<?> ctParameterType, CtTypeRefere
 		return result;
 	}
 
+
+	@Override
+	public boolean hasMethod(CtMethod<?> method) {
+		if (method == null) {
+			return false;
+		}
+
+		// Checking whether the parent is the calling type.
+		try {
+			if (method.getParent() == this) {
+				return true;
+			}
+		} catch (ParentNotInitializedException ex) {
+			return false;
+		}
+
+		// If the method is not the type and, however, a method with its signature exists, this means
```

Comment:
I'm not sure about this part which seems quite specific to your use case.

what if we remove it? could you write c.hasMethod() && foo to obtain the same behavior?